api/src/test/java/org/openmrs/module/sync/SyncProgramAttributeTest.java

```diff
@@ -50,6 +52,81 @@ public String getInitialDataset() {
             throw new RuntimeException(e);
         }
 	}
+
+	@Test
+	@Transactional(propagation = Propagation.NOT_SUPPORTED)
+	public void shouldSyncProgramAttributeOfTypeLocation() throws Exception {
+		runSyncTest(new SyncTestHelper() {
+
+			public void runOnChild() throws Exception {
+
+				ProgramAttributeType programAttributeType = programService.getProgramAttributeType(2); // Health Center Location
+				Assertions.assertNotNull(programAttributeType);
+				PatientProgram pp = programService.getPatientProgram(1);
+				Assertions.assertNotNull(pp);
+				Location healthCenterLocation = Context.getLocationService().getLocation("Someplace");
+				Assertions.assertNotNull(healthCenterLocation);
+				PatientProgramAttribute attr = new PatientProgramAttribute();
+				attr.setAttributeType(programAttributeType);
+				attr.setValueReferenceInternal(healthCenterLocation.getUuid());
+				attr.setPatientProgram(pp);
+				pp.addAttribute(attr);
+				programService.savePatientProgram(pp);
+				List<PatientProgramAttribute> activeAttributes = pp.getActiveAttributes(programAttributeType);
+				Assertions.assertEquals(1, activeAttributes.size());
+			}
+
+			@Override
+			public void changedBeingApplied(List<SyncRecord> syncRecords, Record record) throws Exception {
+				super.changedBeingApplied(syncRecords, record);
+			}
+
+			public void runOnParent() throws Exception {
+				PatientProgram pp = programService.getPatientProgram(1);
+				Set<PatientProgramAttribute> attributes = pp.getAttributes();
+				Assertions.assertEquals(1, attributes.size());
+				Location location = Context.getLocationService().getLocationByUuid(attributes.iterator().next().getValueReference());
```

Comment:
Again, here you probably want to get Location as `(Location) attributes().iterator().next().getValue()`